src/lib/zork/zork.ts

```diff
@@ -239,17 +239,39 @@ const onDataChannel = (ctx: Context, channel: any) => {
 
   channel.onclose = () => ctx.log(`[channel ${channelId}] datachannel closed`);
   channel.onerror = (err: any) => ctx.log.error(`[channel ${channelId}] datachannel error: ${err}`);
-  if (ctx.mode === 'get' || channelId === BOOTSTRAP_CHANNEL_ID) {
-    // ondatachannel events don't fire for the peer creating the datachannel.
-    // Since the getter is the one that creates datachannels, we don't expect
-    // this to run when we're the getter, so just close the datachannel. Also
-    // close it if it's the bootstrap datachannel, as it's served its purpose.
-    channel.onopen = () => channel.close();
-  } else {
-    channel.onopen = () => ctx.log(`[channel ${channelId}] datachannel opened`);
-    initSocksSession(ctx, channel);
-    channel.onmessage = (event: any) => ctx.socksSession.handleDataFromSocksClient(event.data);
+
+  // ondatachannel events don't fire for the peer creating the datachannel.
+  // Since the getter is always the one that creates datachannels, a getter
+  // doesn't expect any giver-created datachannels, so just closes any onopen.
+  if (ctx.mode === 'get') {
+    channel.onopen = () => {
+      channel.close();
+      ctx.log.error(`closed unexpected giver-created datachannel`);
+    };
+    return;
   }
+  // Got here -> we're the giver.
+
+  if (channelId === HEARTBEAT_CHANNEL_ID) {  // heartbeat datachannel
+    ctx.socket.end();
+    ctx.log(`closed zork connection, handoff to RTC complete`);
+    numGetters++;
+    ctx.log(`incremented numGetters to ${numGetters}`);
+    ctx.heartbeatTimeoutId = setTimeout(ctx.onHeartbeatTimeout, HEARTBEAT_TIMEOUT_MS);
+    channel.onmessage = (event: any) => {
+      ctx.log(`heartbeat: ${event.data}`);
+      clearTimeout(ctx.heartbeatTimeoutId);
+      ctx.heartbeatTimeoutId = setTimeout(ctx.onHeartbeatTimeout, HEARTBEAT_TIMEOUT_MS);
+    };
+    return;
+  }
```

Comment:
here we take any message on the heartbeat datachannel as a heartbeat, but I can change this to only count it if `event.data === HEARTBEAT_MSG` if that's preferable.